stdlib/REPL/test/TerminalMenus/runtests.jl

```diff
@@ -5,10 +5,21 @@ using REPL.TerminalMenus
 using Test
 
 function simulate_input(expected, menu::TerminalMenus.AbstractMenu, keys...)
-    keydict =  Dict(:up => "\e[A",
-                    :down => "\e[B",
-                    :enter => "\r")
+    keydict = Dict(:up => "\e[A",
+                   :down => "\e[B",
+                   :enter => "\r")
+    vimdict = Dict(:up => "k",
+                   :down => "j",
+                   :enter => " ")
```

Comment:
This seems to be root of all evil. Using `"\x20"`, `Int(' ')` and `32` (same) the tests fail. Using `49` (key code for space bar), `"\e49"`, `"\e31"` (I don't know whether the code should be decimal or hex) and `"\x31"` (shot in the dark) the tests hung up.

I need help.

Reply:
@nick-paul it's probably been a while since you've last looked at this code, but do you have an idea how to fix this? When using a menu, the keys `k`, `j`, and `<space>`  behave as intended. Only the tests are failing. Thanks in advance! 🙂

Reply:
I'm a fool. This hasn't been the problem; I didn't notice that `request` modifies the state stored in `menu`, hence I missed a `deepcopy` a couple of lines further down:

```julia
    errs = []
    got = _simulate_input(keydict, deepcopy(menu), keys...; kwargs...)
    got == expected || push!(errs, :arrows => got)
```